### DISCUSSION TITLE
scatter and update color

### DISCUSSION THREAD

**Discussion by grisha-lu** [pyqtgraph/pyqtgraph#2534] [Q&A] [answered]
Hi!
How to draw dots, as in matplotlib, the figure below is an example?
When you click the refresh color button.
This must be done for 6 graphs with 20,000 points on each.
```
import matplotlib.pyplot as plt
plt.scatter(x, y, c=z, s=1)
```
![Безымянный](https://user-images.githubusercontent.com/96116423/203088556-0a1ec14c-5511-48d8-be66-809509d8c60d.png)

**Comment by ntjess** [comment, marked as answer]
Hi @grisha-lu, you might find the following helpful:

Some good examples of colored scatterplots can be found if you run
```
python -m pyqtgraph.examples
```
and run the GraphicsItems > Scatter Plot example.

For the colorbar, you can refer to the GraphicsItems > ColorBarItem example.

Hope this helps!
